server/src/main/java/io/deephaven/server/plugin/type/JsTypeDistributionFromPackageJsonSystemPropertyRegistration.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.server.plugin.type;
+
+import io.deephaven.plugin.Registration;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Iterator;
+
+/**
+ * A {@link Registration} that sources {@link JsTypeDistribution#fromPackageJson(Path)} from the system property
+ * {@value FROM_PACKAGE_JSON}. Multiple values can be specified by using a comma as a separator.
+ *
+ * <p>
+ * Potentially useful for development purposes while developing an NPM js plugin.
+ */
+public final class JsTypeDistributionFromPackageJsonSystemPropertyRegistration implements Registration {
+
+    private static final String FROM_PACKAGE_JSON =
+            "io.deephaven.server.plugin.type.JsTypeDistribution.fromPackageJson";
```

Comment:
Should just be the directory with a `package.json`, not the path to the `package.json` itself.

Also the description of the PR should be updated to match these code changes.

Reply:
I'm happy to change it to the directory as opposed to the package.json file.

I'll update the PR description once we are happy w/ the proposed naming / functionality.